### PR TITLE
PTFM-17112 - App and applet for remote builds on 14.04 workers

### PR DESCRIPTION
**Pull request by geetduggal**
@psung Please review at your convenience, and thanks in advance!  This pull request basically contains an app and an applet for remote builds on trusty workers and they are used by the companion pull request in the 'dx-toolkit' repo.

### CODE REVIEW

File: tarball_app_builder_trusty/Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2013 DNAnexus, Inc.
```
Comment (psung) [inline tarball_app_builder_trusty/Makefile]: Use the current year in the boilerplate on newly introduced files.

File: tarball_app_builder_trusty/resources/opt/pythonlibs/app_builder.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
+# Copyright (C) 2013 DNAnexus, Inc.
```
Comment (psung) [inline tarball_app_builder_trusty/resources/opt/pythonlibs/app_builder.py]: You probably intended to not check this in and have it just be created by the Makefile. If so, add a pattern to the root .gitignore so such files don't inadvertently get added again.

File: tarball_app_builder_trusty/tarball_app_builder.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2013 DNAnexus, Inc.
```
Comment (psung) [inline tarball_app_builder_trusty/tarball_app_builder.py]: Minor / or todo later: this file seems to be the same as tarball_app_builder, do you want to lift this to the top level as well and have it be copied into both places by the Makefile?

File: tarball_applet_builder_trusty/dxapp.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "tarball_applet_builder_trusty",
+  "title": "Tarball Applet Builder (Trusty)",
+  "summary": "Builds an applet from a tarball",
+  "categories": ["Debugging"],
+  "dxapi": "1.0.0",
+  "runSpec": {"file": "tarball_applet_builder.py", "interpreter": "python2.7", "execDepends": [{"name": "build-essential"}], "distribution": "Ubuntu", "release": "14.04", "systemRequirements": {"*": {"instanceType": "mem1_ssd1_x4"}}},
```
Comment (psung) [inline tarball_applet_builder_trusty/dxapp.json]: Seems like the precise versions use the default instance type, which is mem2_hdd2_x2-- should we also choose a 2-core instance from this class to have it somewhat be comparable?

File: tarball_app_builder_trusty/resources/opt/pythonlibs/app_builder.py (hunk 2 above)
Comment (geetduggal) [inline tarball_app_builder_trusty/resources/opt/pythonlibs/app_builder.py]: Thanks for catching this!  Done.

File: tarball_app_builder_trusty/tarball_app_builder.py (hunk 3 above)
Comment (geetduggal) [inline tarball_app_builder_trusty/tarball_app_builder.py]: Thanks for the suggestion.  Makes sense, and Done.

**Comment by geetduggal** [comment]
Thanks for the comments!  I added a commit to address them.

**Pull request merged**